ansible/roles/prep.dirs/tasks/repo.yml

```diff
@@ -4,38 +4,40 @@
     insertafter: EOF
     line: "include('{{ mock_dir }}/glusterfs-nightly-repo.tpl')"
   with_items:
-    - "{{ mock_dir }}/centos-stream+epel-next-{{ item }}-x86_64.cfg"
-    - "{{ mock_dir }}/rhel-{{ item }}-x86_64.cfg"
+    - "{{ mock_dir }}/centos-stream+epel-next-{{ os_vers }}-x86_64.cfg"
+    - "{{ mock_dir }}/rhel-{{ os_vers }}-x86_64.cfg"
   loop_control:
     loop_var: mock_config
 
 - name: adapt mock config to use additional template for ceph
   block:
-    - name: determine ceph repo base url
-      shell: curl -s "https://shaman.ceph.com/api/search/?project=ceph&distros=centos/{{ item }}/x86_64&flavor=default&ref=main&sha1=latest" | jq -r .[0].url
-      register: get_ceph_repo_url
+    - block:
+      - name: determine ceph repo base url
+        shell: curl -s "https://shaman.ceph.com/api/search/?project=ceph&distros=centos/{{ os_vers }}/x86_64&flavor=default&ref=main&sha1=latest" | jq -r .[0].url
+        register: get_ceph_repo_url
 
-    - set_fact:
-        "ceph_el{{ item }}_repo_base_url": "{{ get_ceph_repo_url.stdout }}"
+      - set_fact:
+          "ceph_base_url": "{{ get_ceph_repo_url.stdout }}"
+      when: not ceph_base_url
 
     - name: substitute and create ceph repo template
       template:
-        src: "{{ mock_dir }}/ceph-el{{ item }}-dev-repo.tpl.j2"
-        dest: "{{ mock_dir }}/ceph-el{{ item }}-dev-repo.tpl"
+        src: "{{ mock_dir }}/ceph-el{{ os_vers }}-repo.tpl.j2"
```

Comment:
Did you mean _ceph-repo.tpl.j2_?

But then I see a problem. We will then end up with the same base url for every os version template. Let's think on a way out.